upload/catalog/model/extension/smailyforopencart/helper.php

```diff
@@ -87,6 +18,91 @@ public function getSubscribedCustomers($offset, $sync_time) {
+        $this->load->model('setting/setting');
+        $settings = $this->model_setting_setting->getSetting('module_smaily_for_opencart');
+        $subdomain = $settings['module_smaily_for_opencart_subdomain'];
+        $username = $settings['module_smaily_for_opencart_username'];
+        $password = $settings['module_smaily_for_opencart_password'];
+
+        $query = array(
+            'list' => 2,
+            'offset' => $offset,
+            'limit' => 2500,
+        );
+        $unsubscribers = (new \Smaily\Request)
+            ->auth($subdomain, $username, $password)
+            ->setUrlViaEndpoint('contact')
+            ->setData($query)
+            ->get();
+        return $unsubscribers;
+    }
+
+    /**
+     * Send list of subscribers to Smaily.
+     *
+     * @param array $subscribers Subscribers in array format.
+     * @return array $response Response from Smaily.
+     */
+    public function syncSubscribers($subscribers) {
+        $response = array();
+        // Fetch credentials from DB.
+        $this->load->model('setting/setting');
+        $settings = $this->model_setting_setting->getSetting('module_smaily_for_opencart');
+        $subdomain = $settings['module_smaily_for_opencart_subdomain'];
+        $username = $settings['module_smaily_for_opencart_username'];
+        $password = $settings['module_smaily_for_opencart_password'];
+
+        $response = (new \Smaily\Request)
+            ->auth($subdomain, $username, $password)
+            ->setUrlViaEndpoint('contact')
+            ->setData($subscribers)
+            ->post();
+        return $response;
+    }
+
+    /**
+     * Make an abandoned cart API call to Smaily.
+     *
+     * @param string $email Email address of abandoned cart owner.
+     * @return array $response Response from Smaily.
+     */
+    public function sendAbandonedCart($email) {
+        $response = array();
+        // Fetch credentials from DB.
+        $this->load->model('setting/setting');
+        $settings = $this->model_setting_setting->getSetting('module_smaily_for_opencart');
+        $subdomain = $settings['module_smaily_for_opencart_subdomain'];
+        $username = $settings['module_smaily_for_opencart_username'];
+        $password = $settings['module_smaily_for_opencart_password'];
+
+        // Get autoresponder from settings.
+        $autoresponder = html_entity_decode($settings['module_smaily_for_opencart_abandoned_autoresponder']);
+        $autoresponder = json_decode($autoresponder, true);
+        $autoresponder_id = $autoresponder['id'];
+
+        // API call query.
+        $query = array(
+            'autoresponder' => $autoresponder_id,
+            'addresses' => [$email],
+        );
+
+        $response = (new \Smaily\Request)
+            ->auth($subdomain, $username, $password)
+            ->setUrlViaEndpoint('autoresponder')
+            ->setData($query)
+            ->post();
+        return $response;
+    }
```

Comment:
I was wrong when I though bringing in helper function would benefit the code. It actually forces you to load settings twice. It would be better if these API calls exist in methods where they are called from. Can you please move the code back to controllers?

Reply:
Yeah, I agree with you. I kinda knew while coding that we might opt to ditch the helper functions altogether. Will move every helper function in model to where it is used.